crates/burn-tensor/src/tensor/data.rs

```diff
@@ -1,145 +1,522 @@
-            DistributionSamplerKind::Uniform(distribution) => self.rng.sample(distribution),
-            DistributionSamplerKind::Bernoulli(distribution) => {
-                if self.rng.sample(distribution) {
-                    1.elem()
-                } else {
-                    0.elem()
+        for _ in 0..num_elements {
+            data.push(E::random(distribution, rng));
+        }
+
+        TensorData::new(data, shape)
+    }
+
+    /// Populates the data with zeros.
+    pub fn zeros<E: Element, S: Into<Vec<usize>>>(shape: S) -> TensorData {
+        let shape = shape.into();
+        let num_elements = Self::numel(&shape);
+        let mut data = Vec::<E>::with_capacity(num_elements);
+
+        for _ in 0..num_elements {
+            data.push(0.elem());
+        }
+
+        TensorData::new(data, shape)
+    }
+
+    /// Populates the data with ones.
+    pub fn ones<E: Element, S: Into<Vec<usize>>>(shape: S) -> TensorData {
+        let shape = shape.into();
+        let num_elements = Self::numel(&shape);
+        let mut data = Vec::<E>::with_capacity(num_elements);
+
+        for _ in 0..num_elements {
+            data.push(1.elem());
+        }
+
+        TensorData::new(data, shape)
+    }
+
+    /// Populates the data with the given value
+    pub fn full<E: Element, S: Into<Vec<usize>>>(shape: S, fill_value: E) -> TensorData {
+        let shape = shape.into();
+        let num_elements = Self::numel(&shape);
+        let mut data = Vec::<E>::with_capacity(num_elements);
+        for _ in 0..num_elements {
+            data.push(fill_value)
+        }
+
+        TensorData::new(data, shape)
+    }
+
+    /// Converts the data to a different element type.
+    pub fn convert<E: Element>(self) -> Self {
+        if E::dtype() == self.dtype {
+            self
+        } else {
+            TensorData::new(self.iter::<E>().collect(), self.shape)
+        }
+    }
+
+    /// Asserts the data is approximately equal to another data.
+    ///
+    /// # Arguments
+    ///
+    /// * `other` - The other data.
+    /// * `precision` - The precision of the comparison.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the data is not approximately equal.
+    #[track_caller]
+    pub fn assert_approx_eq(&self, other: &Self, precision: usize) {
+        let tolerance = 0.1.pow(precision as f64);
+
+        self.assert_approx_eq_diff(other, tolerance)
+    }
+
+    /// Asserts the data is equal to another data.
+    ///
+    /// # Arguments
+    ///
+    /// * `other` - The other data.
+    /// * `strict` - If true, the data types must the be same.
+    ///              Otherwise, the comparison is done in the current data type.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the data is not equal.
+    #[track_caller]
+    pub fn assert_eq(&self, other: &Self, strict: bool) {
+        if strict {
+            assert_eq!(
+                self.dtype, other.dtype,
```

Comment:
I think in most tests, we want strict to be false, since we can avoid a call to convert this way?

Reply:
Makes sense!